### PR TITLE
feat(www): scroll sidebar to display active doc

### LINKED ISSUE #13602: [docs] Current page not always in view in sidebar when navigating around docs
just a side note: when i jump to a doc then in the doc hierarchy is bolded. 

but i miss here that the bolded selected doc is also moved to be visible area. i always need to scroll down to find the bolded text

_Originally posted by @muescha in https://github.com/gatsbyjs/gatsby/issues/13475#issuecomment-486307738_

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

This is to resolve the discussion from #17447 which has sat open for awhile. To make changes I modified the solution in that PR to:
- use a callback ref (explanation here: https://github.com/gatsbyjs/gatsby/pull/17447#issuecomment-551276995)
- add the ref to section titles to resolve a confusing issue while testing where some docs would be scrolled to (`<Item />`s) and others wouldn't (Accordion section-title`s)
- remove the code for the ScrollPositionProvider that was giving a similar behavior by saving the scroll position as a y coordinate and scrolling back to that location when navigating between top level pages (ie docs/tutorial/contributing) but not for guides within something like the docs

## Demo

You can see in this demo that navigating between pages centers the active doc in the sidebar with `scrollIntoView`. On page reload it is also centered through the same function. The current version of .org loads new pages and opens the active doc in the sidebar but puts the scroll on the top of the sidebar.

scroll

## Related Issues

Fixes #13602

### CODE REVIEW

File: www/src/components/sidebar/item.js [hunk 1]
```diff
@@ -13,61 +13,73 @@ const isItemActive = (activeItemParents, item) => {
...
+}) => {
+  const itemRef = useCallback(
+    async node => {
+      if (item.link === activeItemLink.link && node !== null) {
+        // this noop for whatever reason gives time for React to know what
+        // ref is attached to the node to scroll to it, removing this line
+        // will only scroll to the correct location on a full page refresh,
+        // instead of navigating between pages with the prev/next buttons
+        // or clicking on linking guides or urls
+        await function() {}
```
Comment (gillkyle) [inline www/src/components/sidebar/item.js]: I left this comment to try and explain this but I'm honestly not super sure why this doesn't work without this. I solicited help on this but never heard from anyone else about it so just went with the best solution I could come up with 🙂

File: www/src/components/sidebar/sticky-responsive-sidebar.js [hunk 2]
```diff
@@ -134,9 +131,11 @@ class StickyResponsiveSidebar extends Component {
             <ChevronSvg
               size={16}
               cssProps={{
-                transform: `translate(${iconOffset}px, 5px) rotate(90deg)`,
+                transform: `translate(px, 5px) rotate(90deg)`,
```
Comment (fk) [inline www/src/components/sidebar/sticky-responsive-sidebar.js]: Looks (!) like we lost the `${iconOffset}` here?

**Comment by calcsam** [comment]
This is really great polish to add, well done @gillkyle!

File: www/src/components/sidebar/sticky-responsive-sidebar.js (hunk 2 above)
Comment (gillkyle) [reply www/src/components/sidebar/sticky-responsive-sidebar.js]: Woah, weird, thanks for catching that!

**Pull request merged**